image/image.go

```diff
@@ -75,28 +75,49 @@ func (i *genericImage) Signatures() ([][]byte, error) {
 }
 
 func (i *genericImage) Inspect() (*types.ImageInspectInfo, error) {
-	// TODO(runcom): unused version param for now, default to docker v2-1
-	m, err := i.getSchema1Manifest()
+	m, err := i.getParsedManifest()
 	if err != nil {
 		return nil, err
 	}
-	ms1, ok := m.(*manifestSchema1)
-	if !ok {
-		return nil, fmt.Errorf("error retrivieng manifest schema1")
-	}
-	v1 := &v1Image{}
-	if err := json.Unmarshal([]byte(ms1.History[0].V1Compatibility), v1); err != nil {
-		return nil, err
+	switch t := m.(type) {
+	case *manifestSchema1:
+		v1 := &v1Image{}
+		if err := json.Unmarshal([]byte(t.History[0].V1Compatibility), v1); err != nil {
+			return nil, err
+		}
+		return &types.ImageInspectInfo{
+			Tag:           t.Tag,
+			DockerVersion: v1.DockerVersion,
+			Created:       v1.Created,
+			Labels:        v1.Config.Labels,
+			Architecture:  v1.Architecture,
+			Os:            v1.OS,
+			Layers:        t.GetLayers(),
+		}, nil
+	case *manifestSchema2:
+		rawConfig, _, err := i.src.GetBlob(t.Config.Digest)
+		if err != nil {
+			return nil, err
+		}
+		defer rawConfig.Close()
+		config, err := ioutil.ReadAll(rawConfig)
+		if err != nil {
+			return nil, err
+		}
+		v1 := &v1Image{}
+		if err := json.Unmarshal(config, v1); err != nil {
+			return nil, err
+		}
+		return &types.ImageInspectInfo{
+			DockerVersion: v1.DockerVersion,
```

Comment:
The tag is not a property of an image at all, so it really doesn’t belong in there.  Just document in `types.ImageInspectInfo` that the `Tag` field is legacy fo v2s1 and will not be supported for other formats?